sdk/metricsadvisor/ai-metrics-advisor/test/public/dataSourceCred.spec.ts

```diff
@@ -0,0 +1,287 @@
+          type: "DataLakeGen2SharedKey",
+          accountKey: "account-key"
+        };
+
+        const createdDatalakeCred = await client.createDatasourceCredential(datalakeCred);
+        assert.ok(createdDatalakeCred.id, "Expecting valid datasource credential");
+        createdDatalakeCredId = createdDatalakeCred.id!;
+        assert.equal(createdDatalakeCred.name, datalakeCred.name);
+        assert.equal(createdDatalakeCred.description, datalakeCred.description);
+        assert.equal(createdDatalakeCred.type, datalakeCred.type);
+      });
+
+      it("updates datalake gen2 shared key credential", async function(this: Context) {
+        if (!createdDatalakeCredId) {
+          this.skip();
+        }
+        const dataLakeCredentialPatch = {
+          name: "UpdatedDataLakeCred",
+          description: "updated description",
+          accountKey: "updated account key",
+          type: "DataLakeGen2SharedKey"
+        } as DatasourceCredentialPatch;
+        await client.updateDatasourceCredential(createdDatalakeCredId, dataLakeCredentialPatch);
+        const updated = await client.getDatasourceCredential(createdDatalakeCredId);
+        assert.ok(updated.id, "Expecting valid datasource credential");
+        assert.equal(updated.description, dataLakeCredentialPatch.description);
+        assert.equal(updated.type, dataLakeCredentialPatch.type);
+        assert.equal(updated.name, dataLakeCredentialPatch.name);
+      });
+
+      it("creates service principal credential", async function() {
+        const servicePrincipalCred: ServicePrincipalDatasourceCredential = {
+          ...datasourceCredential,
+          name: "ExampleSPCredential",
+          type: "ServicePrincipal",
+          clientId: "client-id",
+          clientSecret: "client-secret",
+          tenantId: "tenant-id"
+        };
+
+        const createdServicePrincipalCred = await client.createDatasourceCredential(
+          servicePrincipalCred
+        );
+        assert.ok(
+          createdServicePrincipalCred.id,
+          "Expecting valid sql server datasource credential"
+        );
+        createdServicePrincipalCredId = createdServicePrincipalCred.id!;
+        assert.equal(createdServicePrincipalCred.name, servicePrincipalCred.name);
+        assert.equal(createdServicePrincipalCred.description, servicePrincipalCred.description);
+        assert.equal(createdServicePrincipalCred.type, servicePrincipalCred.type);
+      });
+
+      it("updates service principal credential", async function(this: Context) {
+        if (!createdServicePrincipalCredId) {
+          this.skip();
+        }
+        const servicePrincipalCredentialPatch = {
+          name: "UpdatedSPCred",
+          description: "updated description",
+          clientId: "updated-client",
+          clientSecret: "updated-secret",
+          tenantId: "updated-tenant",
+          type: "ServicePrincipal"
+        } as ServicePrincipalDatasourceCredential;
```

Comment:
hmm, we shouldn't cast to `ServicePrincipalDatasourceCredential`?  Now that looking at the sample code, I feel that  it might be better to have individual Patch type, so we can do `const servicePrincipalCredentialPatch: ServicePrincipalCredentialPatch = {` then customer would have better experience with intellisense.  I wonder what IntelliSense would show after you type `type: "ServicePrincipal"` first.

Reply:
Not what I expected, maybe explore a bit more before GA.

Reply:
we changed these to individual patch types and cast to those types